static/box.js

```diff
@@ -1,67 +1,82 @@
-class Shape {
-    /**
-     * Basic shape constructor.
-     * @param x The starting x coordinate.
-     * @param y The starting y coordinate.
-     * @param w The width of the shape.
-     * @param h The height of the shape;
-     * @param fill The color of the shape;
-     */
-    constructor(x, y, w, h, fill) {
-        this.x = x;
-        this.y = y;
-        this.w = w;
-        this.h = h;
-        this.fill = fill;
+
+// This class represents the box at a particular frame of one thing
+class Box {
+
+    constructor(thing, frame, x, y, w, h, interpolated = true) {
+        this.thing = thing; // The parent thing object that owns this keyframe
+        this.frame = frame;
+        this._x = x;
+        this._y = y;
+        this._w = w;
+        this._h = h;
+
+        // true if interpolated, meaning if modified we must insert this back into parent thing as a new keyframe
+        this.interpolated = interpolated;
+    }
+
+    // These getters and setters allow us to insert the keyframe into its parent
+    // thing if it's modified
+    get x() {return this._x;}
+    set x(value) {
+        this.modified();
+        this._x = value;
+    }
+    get y() {return this._y;}
+    set y(value) {
+        this.modified();
+        this._y = value;
     }
+    get w() {return this._w;}
+    set w(value) {
+        this.modified();
+        this._w = value;
+    }
+    get h() {return this._h;}
+    set h(value) {
+        this.modified();
+        this._h = value;
+    }
+
+    modified() {
```

Comment:
Will set be called by the interpolate function to move shapes? For instance, say I interpolate frames 2-99 after manually setting 1 and 100. Will set be called? Since if it will, won't this.interpolatoed always be false?